src/client/state/Notifications.js

```diff
@@ -158,6 +167,34 @@ class Notifications extends EventEmitter {
     [...parentIds].forEach((parentId) => this._deleteNoti(parentId, total, highlight, roomId));
   }
 
+  async _displayPopupNoti(mEvent, room) {
+    if (!settings.showNotifications) return;
+    if (window.Notification === undefined || window.Notification.permission !== 'granted') return;
```

Comment:
if someone deny the `window.Notification.permission` why `settings.showNotifications` flag it as true? may be we should also update  `settings.showNotifications` as soon as user reset the permission so we can have a one source of truth.

Reply:
I thought about a users disabling notifications, removing the permission and adding it again. After that notifications should still disabled in the settings. Maybe however it would be better to enable them again. I'm not sure.

Reply:
it's better to auto enable them again but if they don't grant permission again `settings. showNotifications` still have a true value which is wrong.

Reply:
`settings.showNotifications` is more like a do-not-disturb setting. The only bug that could occur would be sending notifications when `window.Notification.permission` is not `"granted"` which would error out at `new Notification`.
When updating `settings.showNotifications' based on browser-side permissions the value would still be wrong until a new notification arrives (we would have to do the check either way, would not catch the bug above in most circumstances).

Reply:
what exactly `settings.showNotifications` is used for?

Reply:
It's the notification slider in the settings:
![image](https://user-images.githubusercontent.com/35173023/151556757-5ab04c75-f40e-49ec-bea2-de3ddaff6d3f.png)

When the user has not given Cinny notification permissions the button shown below. This however does not change the `settings.showNotifications`.
![image](https://user-images.githubusercontent.com/35173023/151557135-ad63b845-1f92-48bf-8947-4c94f80dd7de.png)

Reply:
As per my understanding the `settings.showNotifications` flag helps to turn ON/OFF desktop notification.
1. When it is OFF it will flag us to not show desktop notifications (even when browser has `window.Notification.permission === granted` ).
2. But when it's ON it flag us to show desktop notification (even when browser has `window.Notification.permission === denied` ). And this is coded as default case + when Notificaiton.permission is removed by browser. It's the another think that if above case come user will see blue Request permission button instead of toggle but under the hood it's still on.
3. The above case can lead to bug if some third person build something on `settings.showNotifications` considering it as a pure flag to user choice of desktop notification.

We may consider it false by default (if `Notification.permission === denied`) else otherwise and we must set it to false if `Notification.permission` gets denied :)

Reply:
There is no real way to check for the permission getting removed. On Chrome and Firefox we can do that. On Safari however there is no live permission access. That would result in the bug described in 3. still occuring until a notification arrives. Would renaming the settings and adding a comment explaining it make sense?

Reply:
For safari the whole desktop notification system doesn't work so we can keep that option to false by default.

For the browser that support it 
1. by default in `src/client/state/settings.js line 118, 119` we can set it as `window.Notification.permission === 'granted` (if Notification.permission is not support it will be false) 
2. we can put a listener in `src/client/event/settings.js` which will update this if browser permission changes. (src/client/event is a place to listen event and do actions accordingly)

:)

Reply:
Safari should actually have [notifications support](https://caniuse.com/notifications). It only lacks [permission api support](https://caniuse.com/permissions-api) which would be needed if we want to listen to permissions changing instead of checking them when sending a notification.

Reply:
other solution: we can make settings._showNotification private and instead use a get function as public api which will consider window.Notification.permission while returning

Reply:
Sounds like a good solution.